### PR TITLE
feat: remove route prop on layout components

### LINKED ISSUE #1282: Remove the need to pass down the route prop in the layout components

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

To make it easier when building out other routes, we needed to remove the required `route` prop. I moved the route generation into a context provider, as well as the active route. Now it will inherit the nested routes from the route provider and display them if the route has any.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #1282 

## Risk

Medium risk as we are moving the logic of routes into a provider, and this may effect plugins.

## Testing

Make sure all routes work, including defi. And test that the plugins are still working as expected.

### CODE REVIEW

File: src/AppProviders.tsx [hunk 1]
```diff
@@ -33,15 +34,17 @@ export function AppProviders({ children }: ProvidersProps) {
...
-                    <PortfolioProvider>
-                      <MarketDataProvider>
-                        <TransactionsProvider>
-                          <ModalProvider>
-                            <DefiManagerProvider>{children}</DefiManagerProvider>
-                          </ModalProvider>
-                        </TransactionsProvider>
-                      </MarketDataProvider>
-                    </PortfolioProvider>
+                    <AppRouteProvider>
```
Comment (cjthompson) [inline src/AppProviders.tsx:37]: How is this different than the `BrowserRouterProvider`? Could they be combined?

**Review by cjthompson — APPROVED** [review]
Approved with comments

**Review by DaoDev44 — APPROVED** [review]

File: src/Routes/Routes.tsx [hunk 2]
```diff
@@ -170,51 +172,69 @@ function useLocationBackground() {
...
+  const currentRoute = useMemo(() => {
+    return appRoutes.find(e => matchPath(location.pathname, { path: e.path, exact: true }))
+  }, [appRoutes, location.pathname])
+
+  return (
+    <RouteContext.Provider value={{ currentRoute, appRoutes }}>{children}</RouteContext.Provider>
+  )
+}
+
+export const useAppRoutes = (): IRouteContext =>
```
Comment (0xdef1cafe) [inline src/Routes/Routes.tsx]: this is dangerous - you're casting something possibly undefined to a concrete type

https://github.com/shapeshift/web/blob/develop/src/context/PluginProvider/PluginProvider.tsx#L135 is how to implement this safely

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

File: src/Routes/Routes.tsx (hunk 2 above)
Comment (0xdef1cafe) [reply src/Routes/Routes.tsx]: otherwise i will gladly give big boy stamp and 🍭

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**